### ISSUE TITLE
Repo initialisation from infrastructure

### ISSUE THREAD

**Issue by manics** [IDR/deployment#1]
Once everything is merged into the infrastructure repo do something like:
```
git remote add infrastructure https://github.com/manics/infrastructure
git fetch infrastructure
git checkout infrastructure/master
git checkout -b filter

git filter-branch --tree-filter '\
    for d in roles roles-dev os-idr-playbooks idr-playbooks; do \
        mkdir -p filter; \
        test -d ansible/$d && mv ansible/$d filter/ || true; \
        done; \
    for d in ansible.cfg requirements.yml; do \
        mkdir -p filter; \
        test -f ansible/$d && mv ansible/$d filter/ || true; \
        done; \
    for d in instance keypairs network security-groups; do \
        mkdir -p filter/roles; \
        test -d ansible/roles/openstack-idr-$d && \
        mv ansible/roles/openstack-idr-$d filter/roles/ || \
        true; \
        done; \
    '

git filter-branch -f --subdirectory-filter filter
```

**Comment by manics** [comment]
Proposed initial repo: https://github.com/manics/deployment/tree/infrastructure-idr-0-3-2

I can think of three options for getting this in:
- Filter and merge in idr-playbooks from https://github.com/IDR/infrastructure/tree/IDR-0.3.2/ansible/idr-playbooks to the existing contents of this repo. The most difficult as it involves merging two separate runs of `git filter-branch`
- Remove the existing contents of this repo in a PR, add the [git filtered `idr-playbooks`](https://github.com/manics/deployment/commit/e65015155743cab2d389e7410c2a326490887fe5) in a subsequent PR
- [Git gilter `idr-playbooks`](https://github.com/manics/deployment/commit/e65015155743cab2d389e7410c2a326490887fe5), and do a one-off force push since this repository has never been useable.

**Comment by manics** [comment]
FYI I've got some molecule tests on the way in https://github.com/manics/deployment/tree/infrastructure-idr-0-3-2-next but it requires more work, so best to test this with the existing (private) playbooks for now.

**Comment by manics** [comment]
https://github.com/manics/deployment/tree/infrastructure-idr-0-3-3 (44ba61c2fbb3183f5fab71bbcc33562cb9c13fd9)
This needs a bump to the requirements, but once that's done I think it can form the first release here.

This is the correspondence between IDR/deployment and openmicroscopy/infrastructure:

Commit https://github.com/manics/deployment/commit/26e68d4716915c6cd73a09efb988cf9a9310e9a5 is corresponds to the 
git-filtered version of https://github.com/openmicroscopy/infrastructure/commit/ad367e6e4f8238e9e223ce094e65943ca51d1e7d

Subsequent commits on deployment:
https://github.com/manics/deployment/compare/26e68d4716915c6cd73a09efb988cf9a9310e9a5...manics:infrastructure-idr-0-3-3

Subsequent commits on infrastructure which will **not** be ported to this repository:
https://github.com/openmicroscopy/infrastructure/compare/ad367e6e4f8238e9e223ce094e65943ca51d1e7d...master

**Comment by manics** [comment]
There's a partial molecule test which uses https://github.com/manics/centos-systemd-ip-docker I don't think it's ready to go into travis yet.

**Comment by joshmoore** [comment]
Steps for testing this:

## On necromancer ##
 * `mkdir /ome/www/downloads.openmicroscopy.org/idr/omero/0.3.3-rc1`
 * `HISTFILE=/dev/null wget https://devspace:******@idr-ci.openmicroscopy.org:8443/job/OMERO-build/lastSuccessfulBuild/artifact/src/target/OMERO.server-5.2.3-414-b95a82e-ice36-b200.zip —no-check-certificate`

## Infrastructure ##
 * `mv vendor vendor.non-idr` - since old versions would not be overwritten.

## management_tools/idr ##
 * `git clone git://github.com/idr/deployment`
 * `ansible-galaxy install -r deployment/ansible/requirements.yml`
 * `export OS_CLOUD=ebi`
 * `ansible-playbook --diff -e idr_environment=test33 deployment/ansible/idr-01-install-idr.yml -e omero_omego_additional_args='"--downloadurl http://downloads.openmicroscopy.org/idr"' -e omero_upgrade=True -e omero_release=0.3.3-rc1`

**Comment by joshmoore** [comment]
In check mode:

```
PLAY RECAP *********************************************************************
test33-a-database          : ok=17   changed=0    unreachable=0    failed=0
test33-a-dockermanager     : ok=32   changed=1    unreachable=0    failed=0
test33-a-omero             : ok=79   changed=6    unreachable=0    failed=0
test33-database            : ok=17   changed=0    unreachable=0    failed=0
test33-management          : ok=7    changed=0    unreachable=0    failed=0
test33-omero               : ok=92   changed=8    unreachable=0    failed=0
test33-proxy               : ok=39   changed=0    unreachable=0    failed=0

```

Note: several of the changes are false-positives from `pip`.

**Comment by joshmoore** [comment]
@manics : is this due to some local config that you have?

```
-# Ansible managed: /Users/spli/work/management_tools/idr/deployment/ansible/vendor/openmicroscopy.rsync-server/templates/etc-rsyncd-conf.j2 modified on 2017-01-17 16:58:21 by spli on ls31908.local
+# Ansible managed
```

**Comment by manics** [comment]
@joshmoore No

**Comment by manics** [comment]
Remaining issues are handled in https://github.com/IDR/deployment/pull/4
